### PR TITLE
Implement constant handling for types of imports.

### PR DESCRIPTION
**Pull request by jonmeow**
This provides support for Const, Pointer, Struct, and Tuple types. It does not cover Class, Function, or Interface which have their own Id and are tracked slightly differently.

### CODE REVIEW

File: toolchain/check/import_ref.cpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      case SemIR::InstKind::StructType:
+        return TryResolveTypedInst(inst.As<SemIR::StructType>());
+
+      case SemIR::InstKind::TupleType:
+        return TryResolveTypedInst(inst.As<SemIR::TupleType>());
+
+      default:
+        context_.TODO(
+            Parse::NodeId::Invalid,
+            llvm::formatv("TryResolveNextInst on {0}", inst.kind()).str());
```
Comment (chandlerc) [inline toolchain/check/import_ref.cpp]: Stale function name here?

File: toolchain/check/import_ref.cpp [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    return context_.types().GetConstantId(
+        context_.GetStructType(context_.inst_blocks().Add(fields)));
+  }
+
+  auto TryResolveTypedInst(SemIR::TupleType inst) -> SemIR::ConstantId {
+    CARBON_CHECK(inst.type_id == SemIR::TypeId::TypeType);
+
+    // First verify that all used types are ready.
+    bool added_todos = false;
+    auto orig_el_type_ids = import_ir_.type_blocks().Get(inst.elements_id);
```
Comment (chandlerc) [inline toolchain/check/import_ref.cpp]: Minor, but `el` as an abbreviation tripped me up for a bit reading this.

File: toolchain/check/import_ref.cpp [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+      el_type_ids.push_back(context_.GetTypeIdForTypeConstant(el_const_id));
+    }
+
+    return context_.types().GetConstantId(context_.GetTupleType(el_type_ids));
+  }
+
+  Context& context_;
+  const SemIR::File& import_ir_;
+  SemIR::ConstantValueStore& import_ir_constant_values_;
+  llvm::SmallVector<SemIR::InstId> todo_;
```
Comment (chandlerc) [inline toolchain/check/import_ref.cpp]: Particularly given the use of a to-do API elsewhere, id prefer a different name. My canonical name for this pattern is "worklist".

Comment (jonmeow) [reply toolchain/check/import_ref.cpp]: WDYT of `stack_` to better label the access pattern?

File: toolchain/check/import_ref.cpp (hunk 2 above)
Comment (jonmeow) [reply toolchain/check/import_ref.cpp]: `elem_`?

File: toolchain/check/import_ref.cpp (hunk 1 above)
Comment (jonmeow) [reply toolchain/check/import_ref.cpp]: Yeah, fixed.

File: toolchain/check/import_ref.cpp (hunk 3 above)
Comment (chandlerc) [reply toolchain/check/import_ref.cpp]: (as discussed, let's go with `work_stack_`)

Comment (jonmeow) [reply toolchain/check/import_ref.cpp]: Done

**Review by chandlerc — APPROVED** [review]
LG!

**Pull request merged**